### PR TITLE
🔀 Fix/store uni v3 position

### PR DESCRIPTION
**Pull request by Thomas-Smets**
Still have to do some gas optimisations, feel free to propose solutions.

### CODE REVIEW

**Comment by Thomas-Smets** [comment]
Failing test is fixed in: #281

**Review by j-vp — APPROVED** [review]
gtg for final review by nethermind

File: src/PricingModules/UniswapV3/UniswapV3PricingModule.sol [hunk 1]
```diff
@@ -282,6 +321,19 @@ contract UniswapV3PricingModule is PricingModule {
         (,, address token0, address token1, uint24 fee, int24 tickLower, int24 tickUpper, uint128 liquidity,,,,) =
             INonfungiblePositionManager(asset).positions(assetId);
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/PricingModules/UniswapV3/UniswapV3PricingModule.sol:322]: True, but we know for sure that the asset is not deposited -> we would read from storage and do extra checks that could be avoided.

File: src/PricingModules/UniswapV3/UniswapV3PricingModule.sol [hunk 2]
```diff
@@ -192,6 +200,37 @@ contract UniswapV3PricingModule is PricingModule {
...
+     * @param asset The contract address of the asset.
+     * @param id The Id of the asset.
+     * @return token0 Token0 of the Liquidity Pool.
+     * @return token1 Token1 of the Liquidity Pool.
+     * @return tickLower The lower tick of the liquidity position.
+     * @return tickUpper The upper tick of the liquidity position.
+     * @return liquidity The liquidity per tick of the liquidity position.
+     */
+    function _getPosition(address asset, uint256 id)
+        internal
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/PricingModules/UniswapV3/UniswapV3PricingModule.sol:214]: Can still be done via de mainregistry.

**Pull request merged**